### PR TITLE
Recommendation History

### PR DESCRIPTION
**Pull request by swastikanata**

### CODE REVIEW

**Review by catherineeangel — COMMENTED** [review]
for the ai suggestions, can you provide keywords that will allow user to use it for search purpose? what do you think?

**Comment by swastikanata** [comment]
> for the ai suggestions, can you provide keywords that will allow user to use it for search purpose? what do you think?

I think this is something that can be implemented on week 3

**Review by emiriko — APPROVED** [review]
I really like the code, it looks clean and not repetitive. Each class also have their own responsibility hence implementing Single Responsibility Principle. Good job!. 

On catherine feedback, I agree with you, I think it could be implemented on week 3.

File: ai/views.py
Comment (carlenee) [inline ai/views.py:1]: Error handling seems adequate for basic scenarios. However, more robust error handling could be implemented, especially when interacting with external services like the OpenAI API. It's essential to anticipate and handle potential errors gracefully. You can consider adding more specific error handling logic, especially around API interactions

**Review by JohannesSetiawan — APPROVED** [review]
The code looks good and the testing seems legit. Good use of Single Responsibility Principle. I agree with the other reviews that the keywords implementation can be implemented later.

**Pull request merged**